pkg/inputs/helper/mysql_common.go

```diff
@@ -0,0 +1,59 @@
+package helper
+
+import (
+	"fmt"
+
+	"github.com/json-iterator/go"
+	"github.com/juju/errors"
+	"github.com/moiot/gravity/pkg/utils"
+)
+
+var myJson = jsoniter.Config{SortMapKeys: true}.Froze()
+
+type SourceProbeCfg struct {
+	SourceMySQL *utils.DBConfig `mapstructure:"mysql"json:"mysql"`
+	Annotation  string          `mapstructure:"annotation"json:"annotation"`
+}
+
+type BinlogPositionsValue struct {
+	CurrentPosition *utils.MySQLBinlogPosition `json:"current_position"`
+	StartPosition   *utils.MySQLBinlogPosition `json:"start_position"`
+}
+
+func SerializeBinlogPositionValue(position *BinlogPositionsValue) (string, error) {
+	s, err := myJson.MarshalToString(position)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	return s, nil
+}
+
+func DeserializeBinlogPositionValue(value string) (*BinlogPositionsValue, error) {
```

Comment:
panic on error is easier I think. Caller can do nothing than exist

Reply:
it's better for the caller to exit, so that we can know the caller's stack trace when error happens

Reply:
if we do a panic rather than fatal, the stack is also kept.